tools/tsp-client/src/fs.ts

```diff
@@ -0,0 +1,49 @@
+import { mkdir, rm, writeFile, mkdtemp, stat, readFile } from "node:fs/promises";
+import { tmpdir } from "node:os";
+import { FileTreeResult } from "./fileTree.js";
+import * as path from "node:path";
+import { Logger } from "./log.js";
+import { parse as parseYaml } from "yaml";
+
+export async function ensureDirectory(path: string) {
+  await mkdir(path, { recursive: true });
+}
+
+export async function removeDirectory(path: string) {
+  await rm(path, { recursive: true, force: true });
+}
+
+export async function createTempDirectory(): Promise<string> {
+  const tempRoot = await mkdtemp(path.join(tmpdir(), "tsp-client-"));
+  Logger.debug(`Created temporary working directory ${tempRoot}`);
+  return tempRoot;
+}
+
+export async function writeFileTree(rootDir: string, files: FileTreeResult["files"]) {
+  for (const [relativeFilePath, contents] of files) {
+    const filePath = path.join(rootDir, relativeFilePath);
+    await ensureDirectory(path.dirname(filePath));
+    Logger.debug(`writing ${filePath}`);
+    await writeFile(filePath, contents);
+  }
+}
+
+export async function tryReadTspLocation(rootDir: string): Promise<string | undefined> {
+  try {
+    const yamlPath = path.resolve(rootDir, "tsp-location.yaml");
+    const fileStat = await stat(yamlPath);
+    if (fileStat.isFile()) {
+      const fileContents = await readFile(yamlPath, "utf8");
+      const locationYaml = parseYaml(fileContents);
+      const { directory, commit, repo } = locationYaml;
```

Comment:
Is `additionalDirectories` removed from the schema of `tsp-location.yaml`? How are the referenced typespec identified when generate sdk code at sdk repository?

Reply:
Given that this tool is analyzing the imports we might not need the additionalDirectories any longer but I wouldn't remove that until we have verification that this is working as expected.

Reply:
ah since openai wasn't using this feature I didn't write support for it and as Wes mentions, it's technically only needed for clone support, but since we want to support that we can fish it out of here as well.

Reply:
That works if this tool can analyze the entry tsp file then fetch the referenced typespec from github.